ProcessMaker/Http/Controllers/Api/UserController.php

```diff
@@ -293,6 +293,12 @@ public function update(User $user, Request $request)
 
         $request->validate(User::rules($user));
         $fields = $request->json()->all();
+        if (($fields['username'] !== $user->getAttribute('username') || isset($fields['password'])) && 
```

Comment:
Please add the corresponding test to validate the following:
- If is an admin user can edit username and password himself
- If is an admin user can edit username and password for other user
- If is a user without permission can not edit
- If is a user with permission can edit himself